### PR TITLE
DEVOPS-1181 lighthouse integration

### PR DESCRIPTION
**Pull request by g-foster2020**

### CODE REVIEW

File: src/_base/docker/image/lighthouse/root/app/run.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+reportDir="/home/headless/results"
+
+mkdir -p "${reportDir}"
+
+lighthouse --no-enable-error-reporting \
+           --chrome-flags="--headless --no-sandbox=true" \
+           --output-path=/home/headless/results/lighthouse-results.html \
+           ${TARGET_URL}
```
Comment (kierenevans) [inline src/_base/docker/image/lighthouse/root/app/run.sh]: shellcheck is complaining that TARGET_URL needs double quoting.

I'd also add `set -e -o pipefail` to after the shebang line to exit on any line erroring.

File: src/_base/harness/attributes/common.yml [hunk 2]
```diff
@@ -157,6 +157,10 @@ attributes.default:
     default_branch: = @('git.main_branch')
     # potentially deprecated, unless it is repurposed
     main_branch: develop
+  
+  lighthouse:
+    target:
+      url: ~
```
Comment (kierenevans) [inline src/_base/harness/attributes/common.yml]: Should the default be `= @('hostname')` to pick up on the main URL for the project?

File: src/_base/docker/image/lighthouse/root/app/run.sh (hunk 1 above)
Comment (g-foster2020) [reply src/_base/docker/image/lighthouse/root/app/run.sh]: Improved and re-tested.

File: src/_base/harness/attributes/common.yml (hunk 2 above)
Comment (g-foster2020) [reply src/_base/harness/attributes/common.yml]: Not sure. What do you think @dcole-inviqa ?

Comment (coledave) [reply src/_base/harness/attributes/common.yml]: seems like a sensible default :)

**Review by kierenevans — DISMISSED** [review]

**Review by kierenevans — APPROVED** [review]

**Pull request merged**